src/main/java/eu/darkbot/api/entities/Entity.java

```diff
@@ -0,0 +1,58 @@
+package eu.darkbot.api.entities;
+
+import eu.darkbot.api.entities.utils.EntityEffect;
+import eu.darkbot.api.objects.LocationInfo;
+
+public interface Entity {
+    /**
+     * @return id of entity
+     */
+    int getId();
+
+    /**
+     * @return memory address
+     */
+    long getAddress();
+
+    /**
+     * Checks that entity is removed.
+     * If (isRemoved() == true) then this entity wont be updated anymore.
+     */
+    boolean isRemoved();
+
+    /**
+     * @return {@link LocationInfo}
+     */
+    LocationInfo getLocationInfo();
+
+    /**
+     * Checks that entity have effect
+     * @param effect to check
+     * @return true if current entity have indicated effect
+     */
+    boolean hasEffect(EntityEffect effect);
```

Comment:
```suggestion
    default boolean hasEffect(EntityEffect effect) {
        return hasEffect(effect,getId());
    }
```
it'd be simpler if it's defined here as default, can apply to a few things